### PR TITLE
#3513 Implement the node list in the editWorkspace page

### PR DESCRIPTION
**Pull request by eechih**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3513 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/Workspace/Edit/MoreActions.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * limitations under the License.
+ */
+
+import React from 'react';
+import IconButton from '@material-ui/core/IconButton';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import MoreIcon from '@material-ui/icons/MoreVert';
+
+MoreActions.propTypes = {};
```
Comment (konekoya) [inline ohara-manager/client/src/components/Workspace/Edit/MoreActions.js]: Redundant proptype?

File: ohara-manager/client/src/components/Workspace/Edit/RestartIndicatorStyles.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    .MuiSnackbarContent-root {
+      background-color: ${theme.palette.warning.main};
+      .message {
+        margin-left: ${theme.spacing(1)}px;
+      }
+    }
+
+    .MuiButton-root {
+      color: white;
```
Comment (konekoya) [inline ohara-manager/client/src/components/Workspace/Edit/RestartIndicatorStyles.js]: There's also a color for this `theme.palette.common.white` :)

File: ohara-manager/client/src/components/Workspace/Edit/MoreActions.js (hunk 1 above)
Comment (eechih) [reply ohara-manager/client/src/components/Workspace/Edit/MoreActions.js]: Yes, I will remove it.

File: ohara-manager/client/src/components/Workspace/Edit/Settings/Nodes/NodeCard.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+const NodeCard = ({ node }) => {
+  const { currentWorker, currentBroker, currentZookeeper } = useWorkspace();
+
+  const [expanded, setExpanded] = React.useState(false);
+  const handleExpandClick = () => {
+    setExpanded(!expanded);
+  };
+
+  const flatServices = services => {
```
Comment (konekoya) [inline ohara-manager/client/src/components/Workspace/Edit/Settings/Nodes/NodeCard.js]: The verb form of the word _flat_ :)

```suggestion
  const flattenServices = services => {
```

File: ohara-manager/client/src/components/Workspace/Edit/RestartIndicatorStyles.js (hunk 2 above)
Comment (eechih) [reply ohara-manager/client/src/components/Workspace/Edit/RestartIndicatorStyles.js]: Thanks~ I will fix it  :)

File: ohara-manager/client/src/context/workspace/EditWorkspaceContext.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+  const calcDirties = cluster => {
+    const settings = get(cluster, 'settings');
+    const stagingSettings = get(cluster, 'stagingSettings');
+
+    const nodes = get(settings, 'nodeNames');
+    const stagingNodes = get(stagingSettings, 'nodeNames');
+
+    return {
+      nodes: size(xor(nodes, stagingNodes)),
```
Comment (konekoya) [inline ohara-manager/client/src/context/workspace/EditWorkspaceContext.js]: 👍  👍  👍

File: ohara-manager/client/src/context/worker/workerActions.js [hunk 5]
```diff
@@ -30,20 +34,29 @@ const fetchWorkersCreator = (
...
 
+  const workers = result.data;
   const workerInfos = await Promise.all(
     map(workers, async worker => {
       const params = pick(worker.settings, ['name', 'group']);
       const result = await inspectApi.getWorkerInfo(params);
       const workerInfo = result.errors ? {} : result.data;
-      return { ...worker, ...workerInfo };
+
+      const result2 = await objectApi.get(params);
```
Comment (konekoya) [inline ohara-manager/client/src/context/worker/workerActions.js]: Maybe come up with better variable names instead of `result` and `result2` :)

Comment (eechih) [reply ohara-manager/client/src/context/worker/workerActions.js]: I will fix it by the way in other PR :)

**Pull request merged**